projects/igniteui-angular/src/lib/grids/grid-public-row.ts

```diff
@@ -28,15 +27,7 @@ export class IgxGridRow implements RowType {
      * ```
      */
     public get data(): any {
-        const grid = this.grid;
-        let allRowsData = grid.allRowsData;
-        if (!this._data) {
-            if (grid.pinnedRecordsCount) {
-                allRowsData = grid.pinning.rows !== RowPinningPosition.Bottom ?
-                    [...grid.pinnedRecords, ...grid.allRowsData] : [...grid.allRowsData, ...grid.pinnedRecords];
-            }
-        }
-        return this._data ?? allRowsData[this.index];
+        return this._data ?? this.grid.data[this.index];

```

Comment:
I guess this is a typo, but this should be `this.grid.dataView[this.index]` right?